fastlane/lib/fastlane/actions/appetize_url_generator.rb

```diff
@@ -0,0 +1,107 @@
+module Fastlane
+  module Actions
+    module SharedValues
+    end
+
+    class AppetizeUrlGeneratorAction < Action
+      def self.run(params)
+        link = "https://appetize.io/embed/#{params[:public_key]}"
+
+        url_params = []
+        url_params << "autoplay=true"
+        url_params << "orientation=#{params[:orientation]}"
+        url_params << "device=#{params[:device]}"
+        url_params << "deviceColor=#{params[:color]}"
+        url_params << "scale=#{params[:scale]}"
+        url_params << "launchUrl=#{params[:launch_url]}" if params[:launch_url]
+        url_params << "language=#{params[:language]}" if params[:language]
+
+        return link + "?" + url_params.join("&")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Generate an URL for appetize simulator"
+      end
+
+      def self.details
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :public_key,
+                                       env_name: "APPETIZE_PUBLICKEY",
+                                       description: "Public key of the app you wish to update",
+                                       is_string: true,
+                                       default_value: Actions.lane_context[SharedValues::APPETIZE_PUBLIC_KEY],
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         if value.start_with?("private_")
+                                           UI.user_error!("You provided a private key to appetize, please provide the public key")
+                                         end
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :device,
+                                       env_name: "APPETIZE_URL_GENERATOR_DEVICE",
+                                       description: "Device type: iphone4s, iphone5s, iphone6, iphone6plus, ipadair, iphone6s, iphone6splus, ipadair2, nexus5, nexus7 or nexus9",
+                                       is_string: true,
+                                       default_value: "iphone5s"),
+          FastlaneCore::ConfigItem.new(key: :scale,
+                                       env_name: "APPETIZE_URL_GENERATOR_SCALE",
+                                       description: "Scale of the simulator",
+                                       is_string: true,
+                                       default_value: "100",
+                                       verify_block: proc do |value|
+                                         available = ["25", "50", "75", "100"]
+                                         UI.user_error!("Invalid scale, available: #{available.join(', ')}") unless available.include?(value)
```

Comment:
for consistency, why do we validate these (if we don't want to validate the device types)